xmake/modules/net/http/download.lua

```diff
@@ -113,6 +113,10 @@ function _curl_download(tool, url, outputfile, opt)
     table.insert(argv, outputfile)
 
     -- download it
+    if opt.try then
+        local ok = try {function () os.vrunv(tool.program, argv); return true end}
+        return ok == nil and false or true
```

Comment:
why? we need not add it. you can use 

```lua
local ok = try {function () http.download(...); return true end}
```

like this

https://github.com/xmake-io/xmake/blob/23f5f45fdbc266eee5fb295641ed6b1703a2111c/xmake/modules/private/action/require/impl/actions/download.lua#L317

Reply:
It's not possible to add a try parameter to all api's, so please use try/catch {} whenever possible.